### PR TITLE
update gethighentropyvalues client hints options and example

### PR DESCRIPTION
**Pull request by Tanych**
#### Summary
As we are going to deprecate the existing client hint `Sec-CH-UA-Full-Version`, using `Sec-CH-UA-Full-Version-List` instead.
This change is going to update the document for Javascript API  [NavigatorUAData.getHighEntropyValues()](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorUAData/getHighEntropyValues) to return a list of brand information containing browser name and full version number. 

#### Motivation
This change helps our long-term goal to depreacate `uaFullVersion`, it helps users to start use `fullVersionList `.

#### Supporting details
* https://wicg.github.io/ua-client-hints/#interface
* https://bugs.chromium.org/p/chromium/issues/detail?id=1260418
* https://github.com/mdn/content/pull/10906
* https://user-agent-client-hints.glitch.me/javascript.html

#### Related issues
WICG/ua-client-hints#196.

#### Metadata

- [ ] Adds a new document
- [x] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/api/navigatoruadata/gethighentropyvalues/index.md [hunk 1]
```diff
@@ -45,8 +46,10 @@ An object containing some or all of the following values, based on the hints req
   - : A {{domxref("DOMString","string")}} containing the device model. For example, `"Pixel 2XL"`.
 - `platformVersion`
   - : A {{domxref("DOMString","string")}} containing the platform version. For example, `"10.0"`.
-- `uaFullVersion`
+- [**Deprecated**]`uaFullVersion`
```
Comment (hamishwillee) [inline files/en-us/web/api/navigatoruadata/gethighentropyvalues/index.md:49]:
```suggestion
- `uaFullVersion` {{deprecated_inline}}
```

File: files/en-us/web/api/navigatoruadata/gethighentropyvalues/index.md [hunk 2]
```diff
@@ -31,7 +31,8 @@ NavigatorUAData.getHighEntropyValues(hints);
     - `"bitness"`
     - `"model"`
     - `"platformVersion"`
-    - `"uaFullVersion"`
+    - [**Deprecated**]`"uaFullVersion"`
```
Comment (hamishwillee) [inline files/en-us/web/api/navigatoruadata/gethighentropyvalues/index.md:34]:
```suggestion
    - `"uaFullVersion"` {{deprecated_inline}}
```

**Comment by hamishwillee** [comment]
Awesome. I've added some fixes as suggestions, but we might need to do a bit more than this:
1. https://developer.mozilla.org/en-US/docs/Web/API/User-Agent_Client_Hints_API#returning_high_entropy_values
2. Might be worth linking to info about low entropy headers for contrast (though maybe not). For HTTP I link to https://developer.mozilla.org/en-US/docs/Web/HTTP/Client_hints#low_entropy_hints
3. You might want to update the browser compatibility data https://github.com/mdn/browser-compat-data/blob/main/api/NavigatorUAData.json#L99 with subfeatures for the allowed UA strings so these can individually be marked as deprecated when needed. Hope this makes sense - what I mean is that as it is now you can only deprecate the whole method, which gives you no way to indicate that some options might not be supported.
4. FYI I marked the corresponding HTTP headers as deprecated some time ago: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Sec-CH-UA-Full-Version#browser_compatibility

Anyway, this would be good enough to go in with the suggestions applied, but I'd seriously consider the BCD fix.

File: files/en-us/web/api/navigatoruadata/gethighentropyvalues/index.md (hunk 2 above)
Comment (Tanych) [reply files/en-us/web/api/navigatoruadata/gethighentropyvalues/index.md:34]: thanks, will updates

**Comment by Tanych** [comment]
@hamishwillee , thanks for your suggestion. 

> 1. https://developer.mozilla.org/en-US/docs/Web/API/User-Agent_Client_Hints_API#returning_high_entropy_values

we seems already update this doc using `fullVersionList`

>2. Might be worth linking to info about low entropy headers for contrast (though maybe not). For HTTP I link to https://developer.mozilla.org/en-US/docs/Web/HTTP/Client_hints#low_entropy_hints

do you mean link the low entropy hints in the page I am going to update? 

>3. You might want to update the browser compatibility data https://github.com/mdn/browser-compat-data/blob/main/api/NavigatorUAData.json#L99 with subfeatures for the allowed UA strings so these can individually be marked as deprecated when needed. Hope this makes sense - what I mean is that as it is now you can only deprecate the whole method, which gives you no way to indicate that some options might not be supported.

I will try to figure out how to update those with subfeatures.
 
>4. FYI I marked the corresponding HTTP headers as deprecated some time ago: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Sec-CH-UA-Full-Version#browser_compatibility

Thanks let me know.

File: files/en-us/web/api/navigatoruadata/gethighentropyvalues/index.md [hunk 3]
```diff
@@ -13,12 +13,15 @@ browser-compat: api.NavigatorUAData.getHighEntropyValues
...
+> **Note:** The terms _high entropy_ and _low entropy_ refer to the amount of information these values reveal about the browser.
+> The values returned as properties are deemed low entropy, and unlikely to identify a user.
+> The values returned by {{domxref("NavigatorUAData.getHighEntropyValues()")}} could potentially reveal more information.
+> These values are therefore retrieved via a {{jsxref("Promise")}}, allowing time for the browser to request user permission, or make other checks.
 
 ## Syntax
 
 ```js
-NavigatorUAData.getHighEntropyValues(hints);
+getHighEntropyValues(hints)
```
Comment (hamishwillee) [inline files/en-us/web/api/navigatoruadata/gethighentropyvalues/index.md:24]: FYI only, this is how we do syntax now days. Essentially it is just to show the parameters.

File: files/en-us/web/api/navigatoruadata/gethighentropyvalues/index.md [hunk 4]
```diff
@@ -31,11 +34,12 @@ NavigatorUAData.getHighEntropyValues(hints);
...
     - `"model"`
     - `"platformVersion"`
-    - `"uaFullVersion"`
+    - `"uaFullVersion"` {{deprecated_inline}}
+    - `"fullVersionList"`
 
 ### Return value
 
-An object containing some or all of the following values, based on the hints requested:
+A {{jsxref("Promise")}} that resolves to an object containing some or all of the following values (based on the hints requested):
```
Comment (hamishwillee) [inline files/en-us/web/api/navigatoruadata/gethighentropyvalues/index.md:42]: FYI only, this is kind of standard text for a method that returns a promise.

File: files/en-us/web/api/navigatoruadata/gethighentropyvalues/index.md [hunk 5]
```diff
@@ -45,24 +49,28 @@ An object containing some or all of the following values, based on the hints req
...
+- `uaFullVersion` {{deprecated_inline}}
   - : A {{domxref("DOMString","string")}} containing the full browser version. For example, `"91.0.4472.124"`.
+- `fullVersionList`
+  - : An array of brand information containing the browser name and full version.
+    For example, `"Chromium";v="91.0.4472.124","Google Chrome";v="91.0.4472.124"`.
 
 ### Exceptions
 
-- A {{domxref("DOMException")}} `NotAllowedError`
+- `NotAllowedError` {{domxref("DOMException")}}
```
Comment (hamishwillee) [inline files/en-us/web/api/navigatoruadata/gethighentropyvalues/index.md:60]: FYI this is the standard way that an exception is now presented.

**Review by hamishwillee — APPROVED** [review]

**Comment by hamishwillee** [comment]
Hi @Tanych ,
I did some very minor follow on tidy ups to current page style. 

Otherwise I think this is more than good enough and will merge (I reread the information on low/high entropy hints and it looks more than sufficient, so ignore my original comment).
It would be worth adding subfeature to BCD. If you get stuck on that you can ping me. 

Thanks very much for this!

**Comment by Tanych** [comment]
@hamishwillee thanks for your work.

**Pull request merged**